design/one-pager-helm-provider.md

```diff
@@ -0,0 +1,524 @@
+            loggingService: logging.googleapis.com/kubernetes
+            monitoringService: monitoring.googleapis.com/kubernetes
+            networkRef:
+              name: gcp-wpclusters-network
+            location: us-central1
+            locations:
+              - us-central1-a
+          writeConnectionSecretToRef:
+            namespace: crossplane-system
+          reclaimPolicy: Delete
+      patches:
+        - fromFieldPath: "metadata.name"
+          toFieldPath: "metadata.name"
+        - fromFieldPath: "metadata.name"
+          toFieldPath: "spec.writeConnectionSecretToRef.name"
+      connectionDetails:
+        - fromConnectionSecretKey: kubeconfig
+    - base:
+        apiVersion: database.gcp.crossplane.io/v1beta1
+        kind: CloudSQLInstance
+        spec:
+          forProvider:
+            databaseVersion: MYSQL_5_6
+            region: us-central1
+            settings:
+              tier: db-custom-1-3840
+              dataDiskType: PD_SSD
+              ipConfiguration:
+                ipv4Enabled: true
+                authorizedNetworks:
+                  - value: "0.0.0.0/0"
+          writeConnectionSecretToRef:
+            namespace: crossplane-system
+          providerRef:
+            name: gcp-provider
+          reclaimPolicy: Delete
+      patches:
+        - fromFieldPath: "metadata.uid"
+          toFieldPath: "spec.writeConnectionSecretToRef.name"
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-mysql"
+        - fromFieldPath: "spec.parameters.storageGB"
+          toFieldPath: "spec.forProvider.settings.dataDiskSizeGb"
+      connectionDetails:
+        - fromConnectionSecretKey: username
+        - fromConnectionSecretKey: password
+        - fromConnectionSecretKey: endpoint
+        - name: port
+          value: "5432"
+    - base:
+        apiVersion: kubernetes.crossplane.io/v1alpha1
+        kind: Provider
+      patches:
+        - fromFieldPath: "metadata.name"
+          toFieldPath: "spec.credentialsSecretRef.name"       
+    - base:
+        apiVersion: helm.crossplane.io/v1alpha1
+        kind: Release
+        spec:
+          forProvider:
+            chart:
+              name: wordpress
+              repository: https://charts.bitnami.com/bitnami
+            namespace: wordpress
+            values: |
+              mariadb.enabled: false
+              externaldb.enabled: true
+            set:
+            - name: externalDatabase.host
+              valueFrom:
+                secretKeyRef:
+                  key: host
+            - name: externalDatabase.user
+              valueFrom:
+                secretKeyRef:
+                  key: username
+            - name: externalDatabase.password
+              valueFrom:
+                secretKeyRef:
+                  key: password
+            - name: blogName
+          reclaimPolicy: Delete
+          providerSelector:
+            matchControllerRef: true
```

Comment:
I think this is trying to select the Kubernetes provider created in the same composition, right? This functionality (selecting rather than referencing a provider) does not exist today. I'd support introducing it, but I'd prefer to do so for all providers rather than making a special case for the Helm provider.